src/bus/index.spec.js

```diff
@@ -0,0 +1,265 @@
+              }
+            };
+
+            bus = new Bus(sdk, baseRequest);
+            bus._baseSocketUrl = expectedHost;
+
+            promise = bus.connect(expectedOrganization.id);
+          });
+
+          context('when intially opening the websocket', function() {
+            it('gets an api token', function() {
+              return promise.then(() => {
+                expect(sdk.auth.getCurrentApiToken).to.be.calledWith(
+                  'contxtAuth'
+                );
+              });
+            });
+
+            it('stores a copy of the websocket', function() {
+              return promise.then(() => {
+                const socket = bus._sockets[expectedOrganization.id];
+
+                expect(socket).to.be.an.instanceof(Socket);
+                expect(socket._organizationId).to.equal(
+                  expectedOrganization.id
+                );
+                expect(socket._socket.url).to.equal(
+                  `${expectedHost}/organizations/${
+                    expectedOrganization.id
+                  }/stream`
+                );
+              });
+            });
+          });
+
+          context('when the websocket closes', function() {
+            beforeEach(function() {
+              return promise.then(() => {
+                server.close();
+              });
+            });
+
+            it('clears out the stored copy of the websocket when it closes', function() {
+              return promise.then(() => {
+                expect(bus._sockets[expectedOrganization.id]).to.be.null;
+              });
+            });
+          });
+        });
+
+        context(
+          'when unsuccessful at connecting to the message bus',
+          function() {
+            let bus;
+            let expectedApiToken;
+            let promise;
+            let sdk;
+            let socketUrl;
+
+            beforeEach(function() {
+              expectedApiToken = faker.internet.password();
+
+              sdk = {
+                ...baseSdk,
+                auth: {
+                  ...baseSdk.auth,
+                  getCurrentApiToken: this.sandbox
+                    .stub()
+                    .resolves(expectedApiToken)
+                }
+              };
+
+              socketUrl = `wss://${faker.internet.domainName()}`;
+              bus = new Bus(sdk, baseRequest);
+
+              bus._baseSocketUrl = socketUrl;
+
+              promise = bus.connect(expectedOrganization.id);
+            });
+
+            it('gets an api token', function() {
+              return promise.catch(() => {
+                expect(sdk.auth.getCurrentApiToken).to.be.calledWith(
+                  'contxtAuth'
+                );
+              });
+            });
+
+            it('rejects the promise', function() {
+              expect(promise).to.be.rejected;
+            });
+
+            it('triggers an error event', function() {
+              return promise.catch((event) => {
```

Comment:
I don't think this is running at all? I'm not sure how you are "forcing" the error to happen

Reply:
It is running. I'm creating a new socket with a different URL than the server so the connection is failing and throwing an error event

Reply:
Ah okay. What do you think about explicitly sending an error? Why this fails isn't really obvious at all